test/SlowTests/Server/Documents/Queries/Dynamic/Map/MatchingAutoMapIndexesForDynamicQueries.cs

```diff
@@ -13,19 +14,20 @@ namespace SlowTests.Server.Documents.Queries.Dynamic.Map
 {
     public class MatchingAutoMapIndexesForDynamicQueries : RavenLowLevelTestBase
     {
-        private readonly DocumentDatabase _documentDatabase;
-        private readonly DynamicQueryToIndexMatcher _sut;
+        private DocumentDatabase _documentDatabase;
+        private DynamicQueryToIndexMatcher _sut;
 
-        public MatchingAutoMapIndexesForDynamicQueries()
+        public void Initialize([CallerMemberName] string caller = null)
         {
-            _documentDatabase = CreateDocumentDatabase();
+            _documentDatabase = CreateDocumentDatabase(caller: $"{GetType().Name}.{caller}");
```

Comment:
This can generate very long names. Just throw if the 'caller' is null.

Reply:
it is the class name . invoking method it is not too long I have run the tests and didn't get any error about too long path names (on windows) I think it is more informative but if you want I can cut it down to the caller (which is not null since I send it explicitly)